backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/dialect/impl/es56/Elasticsearch56Dialect.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.dialect.impl.es56;
+
+import org.hibernate.search.backend.elasticsearch.dialect.impl.ElasticsearchDialect;
+import org.hibernate.search.backend.elasticsearch.dialect.impl.es6.Elasticsearch6Dialect;
+
+/**
+ * The dialect for Elasticsearch 5.6.
+ */
+public class Elasticsearch56Dialect extends Elasticsearch6Dialect implements ElasticsearchDialect {
```

Comment:
Oh, you chose this way. ORM does the opposite. Is there a particular reason, it doesn't seem very natural?

Especially, you could introduce issues in an older dialect by changing things in the newer one. Doesn't look like the best behavior.

Reply:
OK. Can't say I'm excited about this idea because you're right: the last dialect will be the most tested so it's where we don't risk any issue.

Whereas the old ones won't be and we put them at risk.

In any case, I understand where you're coming from so it's not a blocker for me. Just be prepared for my "I told you so" at the first regression of this kind :).